src/software/ai/hl/stp/play/ball_placement_play_test.py

```diff
@@ -12,90 +12,99 @@
+        )
+        simulated_test_runner.gamecontroller.send_ci_input(
+            gc_command=Command.Type.FORCE_START, team=Team.BLUE
+        )
+        # Pass in placement point here - not required for all play tests
+        simulated_test_runner.gamecontroller.send_ci_input(
+            gc_command=Command.Type.BALL_PLACEMENT,
+            team=Team.BLUE,
+            final_ball_placement_point=ball_final_pos,
         )
 
-    # Create world state
-    simulated_test_runner.simulator_proto_unix_io.send_proto(
-        WorldState,
-        create_world_state(
-            yellow_robot_locations=yellow_bots,
-            blue_robot_locations=blue_bots,
-            ball_location=ball_initial_pos,
-            ball_velocity=tbots.Vector(0, 0),
-        ),
-    )
-
-    # Always Validation
-    always_validation_sequence_set = [[]]
-
-    # Eventually Validation
-    eventually_validation_sequence_set = [
-        [
-            # Ball should arrive within 5cm of placement point
-            BallEventuallyEntersRegion(regions=[tbots.Circle(ball_final_pos, 0.05)]),
-        ]
-    ]
+        # Force play override here
+        blue_play = Play()
+        blue_play.name = PlayName.BallPlacementPlay
+
+        simulated_test_runner.blue_full_system_proto_unix_io.send_proto(Play, blue_play)
+
+        # We can parametrize running in ai_vs_ai mode
+        if run_enemy_ai:
+            yellow_play = Play()
+            yellow_play.name = PlayName.EnemyBallPlacementPlay
+
+            simulated_test_runner.yellow_full_system_proto_unix_io.send_proto(
+                Play, yellow_play
+            )
+
+        # Create world state
+        simulated_test_runner.simulator_proto_unix_io.send_proto(
+            WorldState,
+            create_world_state(
+                yellow_robot_locations=yellow_bots,
+                blue_robot_locations=blue_bots,
+                ball_location=ball_initial_pos,
+                ball_velocity=tbots.Vector(0, 0),
+            ),
+        )
 
     simulated_test_runner.run_test(
-        eventually_validation_sequence_set=eventually_validation_sequence_set,
-        always_validation_sequence_set=always_validation_sequence_set,
-        test_timeout_s=test_duration,
+        setup=setup,
+        params=[False, False, False, False],
+        inv_always_validation_sequence_set=[[]],
+        inv_eventually_validation_sequence_set=[
+            [
+                # Ball should arrive within 5cm of placement point
+                BallEventuallyEntersRegion(
+                    regions=[tbots.Circle(ball_final_pos, 0.05)]
+                ),
+            ]
+        ],
+        ag_always_validation_sequence_set=[[]],
+        ag_eventually_validation_sequence_set=[
+            [
+                # Ball should arrive within 5cm of placement point
+                BallEventuallyEntersRegion(
+                    regions=[tbots.Circle(ball_final_pos, 0.05)]
+                ),
+            ]
+        ],
```

Comment:
instead of duplicating validation for ag and inv, what do you think of getting aggregate tests also running invariant testing (since in theory, invariants should be less strict than the aggregate criteria)

Reply:
as talked about on saturday, we'll address this in the next PR